### PR TITLE
Support for debounced and bulk uploading of track messages

### PR DESCRIPTION
**Pull request by lbwexler**
- Note no change to the encoding of the preferences at this time.

### CODE REVIEW

**Review by amcclain — APPROVED** [review]
This looks great.  I confess I did not have a chance yet to run it or try and stress test it in any way, but would be happy for us to merge this + the hoist-react change and burn this in ASAP.

**Comment by amcclain** [comment]
For merge with https://github.com/xh/hoist-react/pull/3802

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 ## 24.0-SNAPSHOT - unreleased
 
+* Support bulk tracking messages.  Improve timestamps on tracking messages
```
Comment (amcclain) [inline CHANGELOG.md:5]: Will be a breaking change requiring new hoist-react

Comment (lbwexler) [reply CHANGELOG.md:5]: indeed -- made the decision that this was simpler -- I think its extremely rare that we ever want to take hoist core updates without hoist-react.  Will amend change log.

**Pull request merged**